src/main/java/com/pileproject/drivecommand/model/nxt/NxtMachineSimulator.java

```diff
@@ -0,0 +1,52 @@
+package com.pileproject.drivecommand.model.nxt;
+
+import com.pileproject.drivecommand.machine.device.DeviceType;
+import com.pileproject.drivecommand.machine.device.input.LineSensor;
+import com.pileproject.drivecommand.machine.device.input.SoundSensor;
+import com.pileproject.drivecommand.machine.device.input.TouchSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedLineSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedSoundSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedTouchSensor;
+import com.pileproject.drivecommand.machine.device.output.Motor;
+import com.pileproject.drivecommand.machine.device.output.simulation.SimulatedMotor;
+import com.pileproject.drivecommand.machine.device.port.InputPort;
+import com.pileproject.drivecommand.machine.device.port.OutputPort;
+
+public class NxtMachineSimulator extends NxtMachine {
```

Comment:
In my understanding, MachineBase should be the interface exactly for such purpose when we designed this library. User's program should handle MachineBase in their codes except for the factory class (currently it's NxtController, right?).

> But users of this library might use only one of the machines in this library

I know there is such case, but this library should assume what I mentioned above. Additionally, I'm wondering if there is benefit when users directly use the derived classes like NxtMachine.